code/presentation/mb-client/src/api/masjid/use-masjids.ts

```diff
@@ -11,17 +12,33 @@ export const useMasjids = () => {
   return useQuery({
     queryKey: ['masjids'],
     queryFn: async () => {
-      const result: PostgrestSingleResponse<Masjid[]> = await supabase.masjid.getAll();
+      const result: PostgrestSingleResponse<Masjid[]> =
+        await supabase.masjid.getAll();
 
       const { data, error: _err } = result;
 
       if (data == null) return null;
 
-      const location = await GetLocation.getCurrentPosition({
-        enableHighAccuracy: true,
-        timeout: 60000,
-      });
-      const fromLoc = { lon: location.longitude, lat: location.latitude };
+      let location = null;
+      if (Platform.OS === 'android') {
+        const { status } = await Location.requestForegroundPermissionsAsync();
+        if (status !== 'granted') {
+          console.log('Permission to access location was denied');
+          return null;
+        }
+        location = await Location.getCurrentPositionAsync({
+          accuracy: Location.Accuracy.Highest,
+        });
+      } else {
+        location = await Location.getCurrentPositionAsync({
+          accuracy: Location.Accuracy.Highest,
+        });
+      }
```

Comment:
ios permissions would be handled differently, add a comment to imply the same.

Reply:
Done